### ISSUE TITLE
[BUG] [Stepper] Not changed focus of the current step in controlled mode

### ISSUE THREAD

**Issue by silviyaboteva** [telerik/kendo-react#979]
If you change the steps from outside (for example from the Prev/Next buttons of a form), the focus ring should be applied to the current step.

The demo that reproduces the issue: https://www.telerik.com/kendo-react-ui/components/form/multi-step-form/

Reported in ticket ID: 1523478

**Comment by AmanNigam1912** [comment]
Could you please provide an ETA for this bug?

**Comment by Xizario** [comment]
Shouldn't the focus style applied when the element is focused, rather than selected? It looks to me it is miss usage of the styles.

I see the example from the ticket is:
https://stackblitz.com/edit/react-uuewuq-31dpbg?file=app%2Fmain.jsx
Where the actual correct usage for the required behaviour will be to style by current, not by focused.
https://stackblitz.com/edit/react-uuewuq-ugyl6q?file=app%2Fmain.jsx

**Comment by AmanNigam1912** [comment]
Yeah the usage of current instead of focus help to solve the issue but I am unable to change the dimensions of the circle around the label. I want it to be auto based on the length of the label and not be constant for every label.

**Comment by silviyaboteva** [comment]
I updated the styles to match the requirement: https://stackblitz.com/edit/react-uuewuq-z9pyi8?file=app%2Fmain.jsx.
The border should be applied to the inner element, e.g. with class `step-label`.

**Comment by simonssspirit** [comment]
After more testing, we can confirm that there is an issue indeed with the focus.

Changing the value from outside the component does not move/remove the focus.

In the example, change the value and notice how the focus is still on the old value:

https://stackblitz.com/edit/react-uuewuq-tv92my?file=app%2Fmain.jsx

![image](https://user-images.githubusercontent.com/11087921/145213756-0744f8f8-decc-4e4f-88b6-95b8d60edbfc.png)

**Comment by Xizario** [comment]
@simonssspirit just a clarification, only 1 element should have the focus state(on the whole app). The issue is that the focus is not removed when the component is blurred. 

Other(or probably the same) issue with the focus state is that when I click on the item (2), it gets focused and selected as expected, however when i press right arrow to focus (3) the 2 is still selected.  With some tab-out tab-in I can even get two focus states that are different from the selected:
<img width="443" alt="Screenshot 2021-12-13 at 16 35 08" src="https://user-images.githubusercontent.com/9458606/145831564-7b525368-f675-4072-aa3f-c06f0590d4e3.png">

**Comment by made2burn** [comment]
We are having this issue as well. Is there an ETA on a fix?  It makes this look incorrect when the steps change in another avenue than clicking the step.

**Comment by jiinus** [comment]
I just used the stepper for the first time in a new project and bumped into this. Any ETA for this fix (which btw celebrates it's 2,5 year birthday...)?

**Comment by iadnonov** [comment]
Fixed in dev version v6.1.0-dev.202311021101
